server-common/src/main/java/org/apache/kafka/server/storage/log/FetchParams.java

```diff
@@ -85,7 +85,7 @@ public boolean fetchOnlyLeader() {
     }
 
     public boolean hardMaxBytesLimit() {
-        return requestVersion <= 2;
+        return requestVersion <= 2 && !shareFetchRequest;
```

Comment:
FetchRequest now only supports version 4 and above. It seems that we could remove `requestVersion` and `hardMaxBytesLimit()`

Reply:
While you are add this, could you also remove the reference to `FetchRequest version 3` in `ReplicaManager`?

```
        // For FetchRequest version 3, we replace incomplete message sets with an empty one as consumers can make
        // progress in such cases and don't need to report a `RecordTooLargeException`
```

Reply:
Done and cleaned `hardMaxBytesLimit` from other places.